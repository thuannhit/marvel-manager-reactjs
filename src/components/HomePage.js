import React, {Component} from 'react';

import { Button, ButtonGroup } from 'reactstrap';
import Ex1 from './Exer1';
import Ex2 from './Exer2';
import Ex3 from './Exer3';
import './style.css';

class HomePage extends Component{
    constructor(props){
        super(props);
        this.state={
            sCurrentUI:'None'
        }
    }
    onClickBtn =(oEvent)=>{
        var sButton=oEvent.target.value;
        if (sButton){
            this.setState({sCurrentUI:sButton});
        };
        
    }

    render(){
        let sCurrentUI=this.state.sCurrentUI;
        let showingUI;
        if (sCurrentUI && sCurrentUI==="Ex1"){
            showingUI=<Ex1/>
        } else if (sCurrentUI && sCurrentUI==="Ex2"){
            showingUI=<Ex2/>
        } else if (sCurrentUI && sCurrentUI ==="Ex3"){
            showingUI=<Ex3/>
        }
        return (
            <div className="d-flex flex-column App">
                <ButtonGroup size="lg">
                    <Button value="Ex1" onClick={this.onClickBtn.bind(this)}>Ex. 1</Button>
                    <Button value="Ex2" onClick={this.onClickBtn.bind(this)}>Ex. 2</Button>
                    <Button value="Ex3" onClick={this.onClickBtn.bind(this)}>Ex. 3</Button>
                </ButtonGroup>
                {showingUI}
            </div>
        )
    }
}

export default HomePage;