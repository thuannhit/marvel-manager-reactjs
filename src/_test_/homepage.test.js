import React from 'react';
import HomePage from '../components/HomePage';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';

jest.mock('../components/Exer1', () => () => <div id="Exer1">Exer1</div>);
jest.mock('../components/Exer2', () => () => <div id="Exer2">Exer2</div>);
jest.mock('../components/Exer3', () => () => <div id="Exer3">Exer3</div>);

it('HomePage renders correctly', () => {
    const tree = renderer
        .create(<HomePage/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

describe('Test onClickBtn function', () => {
    it('Test click event', () => {
      const wrapper = shallow(<HomePage />);
      wrapper.instance().onClickBtn({target:{value:'Ex1'}});
      expect(wrapper.instance().state.sCurrentUI).toEqual('Ex1');
    });
  });