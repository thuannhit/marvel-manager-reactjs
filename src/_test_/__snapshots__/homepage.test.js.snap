// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage renders correctly 1`] = `
<div
  className="d-flex flex-column App"
>
  <div
    className="btn-group-lg btn-group"
    role="group"
  >
    <button
      aria-label={null}
      className="btn btn-secondary"
      onClick={[Function]}
      type="button"
      value="Ex1"
    >
      Ex. 1
    </button>
    <button
      aria-label={null}
      className="btn btn-secondary"
      onClick={[Function]}
      type="button"
      value="Ex2"
    >
      Ex. 2
    </button>
    <button
      aria-label={null}
      className="btn btn-secondary"
      onClick={[Function]}
      type="button"
      value="Ex3"
    >
      Ex. 3
    </button>
  </div>
</div>
`;
